services/src/main/java/io/druid/cli/QueryJettyServerInitializer.java

```diff
@@ -63,7 +69,18 @@ public void initialize(Server server, Injector injector)
     final ServletContextHandler root = new ServletContextHandler(ServletContextHandler.SESSIONS);
     root.addServlet(new ServletHolder(new DefaultServlet()), "/*");
 
-    final AuthConfig authConfig = injector.getInstance(AuthConfig.class);
+    // Add LimitRequestsFilter as first in the chain if enabled.
+    if (serverConfig.isEnableRequestLimit()) {
+      Preconditions.checkArgument(
+          serverConfig.getNumThreads() > 1,
+          "numThreads must be > 1 to enable Request Limit Filter."
+      );
+      log.info("Enabling Request Limit Filter with limit [%d].", serverConfig.getNumThreads()-1);
```

Comment:
why `-1`?

Reply:
if you have x threads to handle request concurrently, then you can allow only "x-1" to go through the filter.. xth request needs to be rejected because (x+1)th request wouldn't come to filter and would rather wait on jetty queue.

Reply:
Ah, my brain had this as processing threads, but its actually jetty threads. gotcha

Reply:
added it as a comment.